src/main/java/io/odpf/firehose/filter/JsonFilter.java

```diff
@@ -0,0 +1,163 @@
+package io.odpf.firehose.filter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import io.odpf.firehose.config.KafkaConsumerConfig;
+import io.odpf.firehose.config.enums.FilterDataSourceType;
+import io.odpf.firehose.config.enums.FilterMessageType;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.metrics.Instrumentation;
+import org.apache.commons.lang3.reflect.MethodUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static io.odpf.firehose.config.enums.FilterDataSourceType.KEY;
+import static io.odpf.firehose.config.enums.FilterDataSourceType.NONE;
+import static io.odpf.firehose.config.enums.FilterMessageType.JSON;
+import static io.odpf.firehose.config.enums.FilterMessageType.PROTOBUF;
+
+/**
+ * JSON-based filter to filter protobuf/JSON messages based on rules
+ * defined in a JSON Schema string.
+ */
+public class JsonFilter implements Filter {
+
+    private final ObjectMapper objectMapper;
+    private final FilterDataSourceType filterDataSourceType;
+    private final FilterMessageType messageType;
+    private Method protoParser;
+    private final String filterJsonSchema;
+    private final Instrumentation instrumentation;
+    private JsonSchema schema;
```

Comment:
make these final.

Reply:
cant make protoParser and schema final cause they're initialized inside try-catch

Reply:
should not be initialised inside try-catch